plugins/inputs/kafka_consumer/kafka_consumer.go

```diff
@@ -170,6 +307,27 @@ func (k *KafkaConsumer) startErrorAdder(acc telegraf.Accumulator) {
 func (k *KafkaConsumer) Start(acc telegraf.Accumulator) error {
 	var err error
 
+	// If TopicRegexps is set, add matches to Topics
+	if len(k.TopicRegexps) > 0 {
+		if err := k.refreshTopics(); err != nil {
+			return err
+		}
+		// If, additionally, TopicRefreshInterval is set,
+		// schedule the refresh.
+		if k.TopicRefreshInterval != 0 {
+			tick := time.Duration(k.TopicRefreshInterval)
+			k.Log.Infof("refreshing topics every %s", tick.String())
+			ticker := time.NewTicker(tick)
```

Comment:
You might want to store the `ticker` in your `KafkaConsumer` to stop it in the `Stop` function!?